source/develop/release-management/features/storage/incremental-backup.html.md

```diff
@@ -12,173 +12,421 @@ feature_status: Design
+   unfreeze, and I/O on the guest can continue normally now.  Any write
+   at this point will be tracked by the new active bitmap, and will not
+   be included in the incremental backup.
+
+1. engine changes the backup status to "ready". Once the user detects
+   the change, the user can start transferring data.
+
+### Finalizing backup
+
+When backup succeeds, backup application copied all incremental data
+successfully.
+
+1. user ask to finalize the backup using oVirt API. engine change
+   backup status to "finalizing".
+
+1. engine end the backup using vdsm backup API, which will use libvirt
+   backup API to end the backup.
+
+1. libvirt uses qemu API to delete the temporary bitmaps and end the
+   backup job.
+
+1. engine tears down and deletes the scratch disks created fro the backup
+   using vdsm storage APIs.
 
+1. If backup application specified a cleanup checkpoint, engine deletes
+   all checkpoints older than the specified checkpoint using vdsm
+   checkpoints API. vdsm uses libvirt checkpoints APIs to perform the
+   request.
 
-## Reference
+1. If running the special "backup" stripped down VM, tear down disks the
+   VM disks using the existing storage APIs, and destroy the VM on the
+   backup host.
 
-#### How to connect image via nbd:
+### Checkpoints cleanup
 
-    qemu-nbd -c /dev/nbd0 -f qcow2 image.qcow2
+Checkpoints may be generated by multiple unrelated or partly related
+users, deleting checkpoints cannot be done automatically by every user
+after backup.
 
-#### How to connect to nbd unix socket:
+The backup API should allow specifying some range of checkpoint to
+delete. Probably specifying a checkpoint that should be kept, deleting
+all older snapshots.
 
-    qemu-img create -f qcow2 -b nbd+unix://?socket=/tmp/nbd.sock overlay.qcow2
+#### Use case 1 - running 2 backup solutions in the same time
+
+User is switching from backup solution A to B. During the transition,
+both systems are performing backups of the same data. If one of the
+systems will delete checkpoints after backup, it can prevent incremental
+backup of the other system.
+
+#### Use case 2 - running backups with different schedule
+
+User is configuring both hourly and daily backup. If both jobs delete
+the checkpoints after backup they may prevent the next job from
+completing the next backup.
+
+### Cancelling backup
+
+When backup fail or aborted, backup application did completed to copy
+all data.
+
+1. Delete scratch disks created for this backup.
+2. If the VM is not running, tear down the disks included in the backup.
+
+
+### Engine database changes
+
+XXX Add disk attachment incremental setting.
+
+- New ```vm_backups``` table
+  - fields: id, vm_id, incremental, creation_date, status
+  - status: initializing, ready, transferring, finallizing
+- ```images``` table
+  new field: ```backup_id``` (foreign key ```bakcups:id```)
+  new field: ```backup_url``` (nbd url used in image transfer ticket)
+
```

Comment:
We will have to track checkpoints, since they are not tracked in the image itself. Add table for the checkpoints.

When creating  a backup, we can find the list of checkpoints needed for the backup, and pass this list to vdsm to configure libvirt.